core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -1443,11 +1443,15 @@ private void checkExponentialCnf(int n) {
         "true");
 
     // "a = 1 or a <> 2" could (and should) be simplified to "a <> 2"
-    // but can't do that right now
     checkSimplifyFilter(
         or(eq(aRef, literal1),
             ne(aRef, literal2)),
-        "OR(=(?0.a, 1), <>(?0.a, 2))");
+        "<>(?0.a, 2)");
+
+    checkSimplifyFilter(
```

Comment:
What about adding a clarifying comment as we have in other tests? I.e. `a < 1 or a > 1 could be simplified to a <> 1`

Reply:
good idea. done.